### DISCUSSION TITLE
DDIM generation code

### DISCUSSION THREAD

**Discussion by Ahmad-Omar-Ahsan** [Project-MONAI/GenerativeModels#470] [Q&A] [answered]
Hello,

I am using a DDIM scheduler to generate images from 1000 steps. Unfortunately, the generated images are dark, unlike the training distribution. I have also generated the intermediate steps, but the intermediate steps are filled with noise. 

I have tried using a DDPM scheduler for the reverse process, which gives me the desired output, but the DDIM scheduler doesn't. How do I debug this?

**Comment by marksgraham** [comment, marked as answer]
In my experience, sometimes a DDIM scheduler doesn't work with 1000 steps during sampling, but will work better for fewer timesteps, say 100. So i would try that. I've also found more training is required to get good samples from a DDIM sampler, compared to a DDPM sampler, so make sure your model has trained long enough

**Reply by Ahmad-Omar-Ahsan** [reply]
Is it because of the deterministic nature of DDIM? 

By default, $`\eta`$ is set to 0 in DDIM, making the reverse process deterministic.

My current experiment is an overfitting test, but I don't think generating good-quality samples with DDIM will be possible when training with 100 samples.

**Reply by Ahmad-Omar-Ahsan** [reply]
The other question I had was about the training process, even though the values of the loss stay the same (approximately) at later epochs, how is DDIM able to generate better quality samples when trained longer than DDPM?

**Reply by marksgraham** [reply]
I think very small changes in the loss can lead to improvements in sample quality because the improvements add up as you sample through the whole diffusion chain, so it can be worth training beyond the point that the loss stops visually improving much
